radixdlt/src/integration/java/com/radixdlt/consensus/deterministic/BFTDeterministicTest.java

```diff
@@ -82,19 +85,33 @@ public void start() {
 		nodes.forEach(ControlledBFTNode::start);
 	}
 
-	public void processNextMsg(int toIndex, int fromIndex, Class<?> expectedClass) {
+	/**
+	 * Returns the next message that should be processed
+	 * @return the next message that should be processed
+	 */
+	public ControlledMessage processNextMsg(int toIndex, int fromIndex, Class<?> expectedClass) {
 		ChannelId channelId = new ChannelId(pks.get(fromIndex), pks.get(toIndex));
 		Object msg = network.popNextMessage(channelId);
 		assertThat(msg).isInstanceOf(expectedClass);
 		nodes.get(toIndex).processNext(msg);
+		return new ControlledMessage(channelId.getSender(), channelId.getReceiver(), msg);
```

Comment:
Would rather not use the existing ControlledMessage. Should create a different class for this tests are abstracted from ECPublicKeys and only use indices instead.